src/components/v5/frame/NavigationSidebar/partials/NavigationSidebarButton/types.ts

```diff
@@ -1,10 +1,13 @@
 import { ButtonHTMLAttributes } from 'react';
+import { NavigationSidebarSecondLevelProps } from '../NavigationSidebarSecondLevel/types';
 
 export interface NavigationSidebarButtonProps
   extends ButtonHTMLAttributes<HTMLButtonElement> {
   iconName: string;
   label: string;
   isActive?: boolean;
   isExpanded?: boolean;
+  hideMobile?: boolean;
+  secondLevelMenuProps?: NavigationSidebarSecondLevelProps;
```

Comment:
This prop is redundant here, because you're using it as a boolean and you are passing the whole second level menu object. You should simplify that and just change it to boolean that can be named `withCaretIcon`. By default you can set it to true, so that every button will have it by default. Then inside the component that is using this one, just set it to false when  `secondLevelMenuProps` is undefined.